config/300-ibmmqsource.yaml

```diff
@@ -0,0 +1,225 @@
+              queueManager:
+                description: IBM MQ Queue Manager name.
+                type: string
+              queueName:
+                description: IBM MQ event source Queue name.
+                type: string
+              delivery:
+                description: Events delivery parameters.
+                type: object
+                properties:
+                  deadLetterQueueManager:
+                    description: The name of the Dead Letter Queue Manager.
+                    type: string
+                  deadLetterQueue:
+                    description: The name of the Dead Letter Queue.
+                    type: string
+                  backoffDelay:
+                    description: Delay between delivery retries for failed messages.
+                    type: integer
+                  retry:
+                    description: The number of delivery retries before moving the message to DLQ.
+                    type: integer
+              sink:
+                description: The destination of events sourced from IBM MQ.
+                type: object
+                properties:
+                  ref:
+                    description: Reference to an addressable Kubernetes object to be used as the destination of events.
+                    type: object
+                    properties:
+                      apiVersion:
+                        type: string
+                      kind:
+                        type: string
+                      namespace:
+                        type: string
+                      name:
+                        type: string
+                    required:
+                    - apiVersion
+                    - kind
+                    - name
+                  uri:
+                    description: URI to use as the destination of events.
+                    type: string
+                    format: uri
+                oneOf:
+                - required: [ref]
+                - required: [uri]
+              credentials:
+                type: object
+                properties:
+                  username:
+                    type: object
+                    properties:
+                      value:
+                        description: Plain text user ID.
+                        type: string
+                      valueFromSecret:
+                        description: A reference to a Kubernetes Secret object containing the user ID.
+                        type: object
+                        properties:
+                          name:
+                            type: string
+                          key:
+                            type: string
+                        required:
+                        - name
+                        - key
+                  password:
+                    type: object
+                    properties:
+                      value:
+                        description: Plain text password.
```

Comment:
im sorry if this sounds stupid.. but the plain text password for what? Where do i get this? Or is it user created? What's it being used for?